numba/core/caching.py

```diff
@@ -543,6 +544,10 @@ def _load_index(self):
         Load the cache index and return it as a dictionary (possibly
         empty if cache is empty or obsolete).
         """
+
+        def round_timestamp(stamp: tuple[float, int]) -> tuple[float, int]:
+            return floor(stamp[0]), stamp[1]
```

Comment:
Test fail with

```pytb
File ~/work/1/s/numba/core/caching.py:549, in IndexDataCacheFile._load_index.<locals>.round_timestamp(stamp)
    548 def round_timestamp(stamp: tuple[float, int]) -> tuple[float, int]:
--> 549     return floor(stamp[0]), stamp[1]

TypeError: must be real number, not str
```
(https://dev.azure.com/numba/numba/_build/results?buildId=18207&view=logs&j=c3c1ecc7-c144-5f9e-e339-a9829aa82e09&t=9a2121c1-dd7b-5e3c-d1e5-29828ad339fc&l=10924)

Reply:
This code should not depend on the exact type of `stamp`. The rounding of timestamp should appear at where timestamp is produced (where `os.stat` is called) if the granularity is to be adjusted.